### ISSUE TITLE
500 when subscribing Backend-to-backend application to api-key plan

### ISSUE THREAD

**Issue by Mattue** [gravitee-io/issues#8661]
# Describe the bug

We have 1 API with `api-key` plan, 1 `simple` application, 1 `backend-to-backend` application.

When subscribing to API using `simple` application, it's working fine.

When subscribing to API using `backend-to-backend` application subscription fails with `Cannot invoke "String.toLowerCase()" because "typeId" is null`

# To Reproduce

1.  Create API with `api-key` plan
2.  Create `backend-to-backend` application
3.  Go to `API's -> Your API -> Subscriptions`
4.  Click on `+` sign
5.  Try to subscribe `backend-to-backend` application to `api-key` plan
6.  See error

# Expected behavior

`Backend-to-backend` application subscribes to `api-key` plan and api-key created for application. Or reasonable error message which describes why `backend-to-backend` application cannot be subscribed to `api-key` plan.

# Current behaviour

Subscription is failing with error `Cannot invoke "String.toLowerCase()" because "typeId" is null`

Gravitee API communication example

Request

 POST [https://my-host.host/management/organizations/DEFAULT/environments/DEFAULT/apis/cd2af225-38f9-4fb9-aaf2-2538f91fb946/subscriptions?application=79a2e9c8-a19e-46aa-a2e9-c8a19e96aa2d&plan=81665400-42a6-4057-a654-0042a6c057cf](https://my-host.host/management/organizations/DEFAULT/environments/DEFAULT/apis/cd2af225-38f9-4fb9-aaf2-2538f91fb946/subscriptions?application=79a2e9c8-a19e-46aa-a2e9-c8a19e96aa2d&plan=81665400-42a6-4057-a654-0042a6c057cf)([https://my-host.host/management/organizations/DEFAULT/environments/DEFAULT/apis/cd2af225-38f9-4fb9-aaf2-2538f91fb946/subscriptions?application=79a2e9c8-a19e-46aa-a2e9-c8a19e96aa2d&plan=81665400-42a6-4057-a654-0042a6c057cf](https://my-host.host/management/organizations/DEFAULT/environments/DEFAULT/apis/cd2af225-38f9-4fb9-aaf2-2538f91fb946/subscriptions?application=79a2e9c8-a19e-46aa-a2e9-c8a19e96aa2d&plan=81665400-42a6-4057-a654-0042a6c057cf))

Response

 { "message": "Cannot invoke \"String.toLowerCase()\" because \"typeId\" is null", "http_status": 500 } 

*   Environment: Kubernetes, gravitee apim v3.18.10
*   OS: Ubuntu 18.04.6 TLS
*   Browser: firefox, chrome

**Comment by Mattue** [comment]
In addition to that I'm having same issue when adding backend-to-backend application to a group

```

**Comment by Mattue** [comment]
Just updated to 3.18.18, issue with application subscription does not occurs. Thanks!

But adding application to group is still a problem
![изображение](https://user-images.githubusercontent.com/9050530/215869229-d1d49283-2d92-456d-ba2b-c802655f6522.png)

Request
```
{
    "name": "my-app",
    "description": "client-secred-experimentation",
    "groups": [
        "66e60e5e-5ccb-4ed9-a60e-5e5ccb9ed922"
    ],
    "settings": {
        "oauth": {
            "client_id": "xxx",
            "client_secret": "xxx",
            "redirect_uris": [],
            "response_types": [
                "code",
                "none"
            ],
            "grant_types": [
                "authorization_code",
                "client_credentials"
            ],
            "renew_client_secret_supported": false
        }
    },
    "disable_membership_notifications": false,
    "api_key_mode": "UNSPECIFIED"
}
```
Response
```
{
  "message" : "Cannot invoke \"String.toLowerCase()\" because \"typeId\" is null",
  "http_status" : 500
}
```

Tested on:
   server:
      Environment: Kubernetes, gravitee apim v3.18.18, PostgreSQL 15.1 64-bit
      OS: Ubuntu 18.04.6 TLS
      Browser: firefox, chrome
   local:
      Environment: docker-compose, gravitee apim v3.20.0, PostgreSQL 15.1 64-bit
      OS: Windows 10
      Browser: firefox, chrome

**Comment by Mattue** [comment]
Tried to trace error and find issue in code

Looking this exception:
```
java.lang.NullPointerException: Cannot invoke "String.toLowerCase()" because "typeId" is null
at io.gravitee.rest.api.service.impl.configuration.application.ApplicationTypeServiceImpl.lambda$getApplicationType$0(ApplicationTypeServiceImpl.java:64)
```
found this class - [ApplicationTypeServiceImpl](https://github.com/gravitee-io/gravitee-api-management/blob/master/gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/configuration/application/ApplicationTypeServiceImpl.java), line 60

May be this may help.

**Comment by Mattue** [comment]
@mouligno , @gaetanmaisse, just closed? 😢

**Comment by gaetanmaisse** [comment]
@Mattue we attached it to the wrong version so it's not displayed in the changelog but no, not "just closed", just fixed! 🎉 
it has been fixed and released in APIM 3.19.19 and 3.20.14.
We will update the changelogs accordingly tomorrow 😉

**Comment by Mattue** [comment]
@gaetanmaisse , thank you for reply. Will test shortly!
